Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeActionHandler.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/string/string.h>
+
+#include <AzToolsFramework/ActionManager/ActionManagerRegistrationNotificationBus.h>
+#include <AzToolsFramework/ComponentMode/EditorComponentModeBus.h>
+
+namespace AzToolsFramework
+{
+    class ActionManagerInterface;
+    class MenuManagerInterface;
+    class HotKeyManagerInterface;
+
+    class ComponentModeActionHandler
```

Comment:
update: It would definitely be useful to add an API comment for this new type

Reply:
Done!